### PR TITLE
Speedup `combine_tobac_feats`

### PR DESCRIPTION
**Pull request by freemansw1**
I sped up `utils.general.combine_tobac_feats` several times by getting rid of `.loc` functions. Because I used the `idx` column in the new `combine_tobac_feats` function (as it is guaranteed to be output by feature detection), I had to add it to the testing generation function and fix some of the tests to expect an `idx` column. 

* [x] Have you followed our guidelines in CONTRIBUTING.md? 
* [x] Have you self-reviewed your code and corrected any misspellings? 
* [x] Have you written documentation that is easy to understand?
* [x] Have you written descriptive commit messages? 
* [ ] Have you added NumPy docstrings for newly added functions? 
* [x] Have you formatted your code using black? 
* [x] If you have introduced a new functionality, have you added adequate unit tests?
* [x] Have all tests passed in your local clone? 
* [ ] If you have introduced a new functionality, have you added an example notebook?
* [x] Have you kept your pull request small and limited so that it is easy to review? 
* [x] Have the newest changes from this branch been merged?

### CODE REVIEW

**Comment by freemansw1** [comment]
The CI was failing due to an issue with the version number (we require a `x.x.x` mapping, it was `1.5`). This has been fixed.

**Comment by freemansw1** [comment]
I should have said the speedup - it went from 5min 30s to ~15s on my test dataset of several hundred thousand features.

**Comment by freemansw1** [comment]
I should note that I doubt that this is as efficient as it possibly could be; in particular the `rank` function seems relatively slow for what it does, but I think this is still merge-able given the speed improvements here.

File: tobac/utils/general.py [hunk 1]
```diff
@@ -539,21 +535,13 @@ def combine_tobac_feats(list_of_feats, preserve_old_feat_nums=None):
...
-        curr_row_count = len(combined_df.loc[combined_df["time"] == curr_time])
-        feat_num_arr = np.arange(start_feat_num, start_feat_num + curr_row_count)
-        combined_df.loc[combined_df["time"] == curr_time, "feature"] = feat_num_arr
-        start_feat_num = np.max(feat_num_arr) + 1
-
-    combined_df = combined_df.reset_index(drop=True)
-    return combined_df
+    # count_per_time = combined_feats.groupby('time')['index'].count()
+    combined_df["frame"] = combined_df["time"].rank(method="dense").astype(int) - 1
+    combined_sorted = combined_df.sort_values(["frame", "idx"], ignore_index=True)
```
Comment (w-k-jones) [inline tobac/utils/general.py:544]: **question:** To clarify, idx is assigned as the index of each feature detected per frame in feature detection, and may not be contiguous due to lower threshold features being removed?

**suggestion:** Is sorting by idx necessary? I guess it's to try and keep features on the same frame in the same order as in the pre-combined dataframes?

**Review by w-k-jones — APPROVED** [review]
Looks good, happy for this to be merged. My only comments are regarding the use of idx, and whether this is needed. It might be easier for future tests to have idx automatically generated when producing test data

File: tobac/utils/general.py (hunk 1 above)
Comment (freemansw1) [reply tobac/utils/general.py:544]: Yes, `idx` is assigned as the index of each feature detected per frame in feature detection. 

Yeah, I wanted to preserve as much as possible as the original dataframes. I could see users getting confused if sorting on `feature` and sorting on (`frame`, `idx`) produced different results.

**Review by kelcyno — APPROVED** [review]
I don't have anything to add here, it looks good to me at this time, pending Will's question on idx.

**Comment by w-k-jones** [comment]
Seconding, happy for this to be merged

**Pull request merged**